sunkit_image/tests/test_offlimb.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+import pytest
+import astropy.units as u
+
+import sunpy
+import sunpy.map
+import sunpy.data.test
+
+import sunkit_image.utils.utils as utils
+import sunkit_image.offlimb as off
+
+
+@pytest.fixture
+def map_test1():
+        x = np.linspace(-2, 2, 5)
+        grid = np.meshgrid(x, x.T)
+        test_data1 = np.sqrt(grid[0]**2 + grid[1]**2)
+        test_data1 *= 10
+        test_data1 = 28-test_data1
+        test_data1 = np.round(test_data1)
+        header = {'cunit1': 'arcsec', 'cunit2': 'arcsec'}
+        test_map1 = sunpy.map.Map((test_data1, header))
+        return test_map1
+
+
+@pytest.fixture
+def map_test2():
+        x = np.linspace(-2, 2, 5)
+        grid = np.meshgrid(x, x.T)
+        test_data1 = np.sqrt(grid[0]**2 + grid[1]**2)
+        test_data1 *= 10
+        test_data1 = 28-test_data1
+        test_data1 = np.round(test_data1)
+        header = {'cunit1': 'arcsec', 'cunit2': 'arcsec'}
+        test_data2 = np.where(test_data1[:, 0:2] == 6, 8, test_data1[:, 0:2])
+        test_data2 = np.concatenate((test_data2, test_data1[:, 2:]), axis=1)
+        test_map2 = sunpy.map.Map((test_data2, header))
+        return test_map2
+
+
+@pytest.fixture
+def radial_bin_edges():
+        radial_bins = utils.equally_spaced_bins(inner_value=0.001, outer_value=0.003, nbins=5)
+        radial_bins = radial_bins * u.R_sun
+        return radial_bins
+
+
+def set_attenuation_coefficients(order):
+    attenuation_coefficients = np.zeros((2, order + 1))
+    attenuation_coefficients[0][:] = np.linspace(1, 0, order + 1)
+    attenuation_coefficients[1][:] = np.linspace(1, 0, order + 1)
+    return attenuation_coefficients
+
+
+def test_nrgf(map_test1, map_test2, radial_bin_edges):
+
+    with pytest.warns(Warning, match="Missing metadata for solar radius: assuming photospheric limb as seen from Earth"):
```

Comment:
Since we are providing the metadata during map creation I will remove this line.

Reply:
Well if we raise the warning, we should either fix the cause or suppress it in the tests.

Reply:
I will check again whether the warnings are raised if I remove this line and then get back to you.